[History] HistorySwiper, HistorySlide, YearSlide 컴포넌트 초안

dani-day opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
> 일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

- History 페이지의 기능 및 레이아웃 (header, main, nav, footer 등 공통 section 을 제외) 을 구현하였습니다.

### [피그마 시안]
![historySection](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/76cf5d4e-6e3d-4883-8527-14d3015be701)

### [구현 내용]
**1920**
![스크린샷 2023-12-08 오전 10 52 44](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/e19697f4-bf5d-4d31-a233-a48227a9836e)


**1024**
![스크린샷 2023-12-08 오전 10 52 08](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/0027591c-59e7-4b7a-912b-0d55b5754cfc)


**375**
![스크린샷 2023-12-07 오후 2 58 12](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/351d803e-87db-4f34-9157-64e6bcc05734)


## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

- History 페이지 내 공통 section 을 제외한 연혁 section 기능 및 레이아웃을 구현하였습니다.
- History 페이지 내 연혁 section 을 컴포넌트화 하였습니다.
<img width="615" alt="image" src="https://github.com/ReOpenSG/ReOpenSG/assets/134567470/1e8fd706-e0e4-42fd-beaf-4a6a8d94ce9f">

- 모바일, 태블릿, 데스크탑에서 반응형 레이아웃을 구현하였습니다.
- CSS 모듈화 완료하였습니다.

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

> 없으면 "없음" 이라고 기재해 주세요

- 없음

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

- CSS 모듈화가 팀 의도에 맞게 분리되어 있는지 검토해주시면 감사하겠습니다.

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

- 없음

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

dani-day committed:
⚙️ [Chore]  : Swiper 설치 완료

History 페이지의 연혁 카드 기능을 위해, Swiper 설치 완료하였습니다.

Related to: #6

dani-day committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

dani-day committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

dani-day committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

dani-day committed:
✨ [Feat] : History 페이지의 연혁 카드 기능 추가

History 페이지의 연혁 카드 기능을 추가하였습니다.
+ 연혁 Swiper 와 카드 Swiper 를 추가하였습니다.
+ 연혁 Swiper 와 카드 Swiper 가 연동되어 움직이도록 하였습니다.
+ 첫번째 카드가 가운데 오도록 하였으며, 연혁 Swiper 의 slidesPerView 를 더 많이 보이도록 설정하였습니다.
+ 연혁 Swiper 는 isActive 여부에 따라 다른 마크업 및 스타일링이 적용됩니다. (컬러, 폰트 사이즈)
+ 기본 Markup 과 스타일링을 완료하였습니다.

! 내용은 dummy 로 넣어두었습니다.
! 카드 스타일링이 완료되지 않았습니다. (카드 width 조정 필요, 항목별 왼쪽 원형 아이콘 추가 필요)

dani-day committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

dani-day committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

dani-day committed:
✨ [Feat] : History 페이지의 YearSlide, HistorySlide 컴포넌트 분리, json 데이터 연결

+ YearSlide, HistorySlide 컴포넌트 분리하였습니다.
+ historyData.json 생성하여 해당 json 데이터를 활용하여 연혁 swiper 렌더링되도록 하였습니다.
+ 중복 code 삭제하였습니다. (조건부 렌더링 시, 스타일링만 변경되도록 하여 중복되던 조건부 마크업 코드 삭제)
+ Swiper 에 mousewheel 상호작용 기능 추가하였습니다.
+ 카드 스타일링 일부 수정하였습니다. (카드 배경 및 여백, list-disc 스타일 추가 후 왼쪽 여백 추가)
+ 2023년 회사 소개서에 있는 연혁 데이터로 수정하였습니다.

! 반응형 레이아웃 확인 및 적용 필요 (Mobile, Tablet, Desktop)
! tailwind.config.js 에 추가된 spacing 으로 스타일링 변경 필요

Related to: #6

dani-day committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

dani-day committed:
✨ [Feat] : 반응형 구현 및 CSS 모듈화 (History.module.css)
+ mobile 을 기준으로, tablet 및 desktop 의 반응형을 구현하였습니다.
+ History.jsx, HistorySlide.jsx, YearSlide.jsx 의 CSS 를 History.module.css 에 모듈화하였습니다.
+ postcss.config.js 에서 tailwindcss/nesting 관련 설정을 추가하였습니다.

Related to: #6

dani-day committed:
♻️ [Refactor] : History 페이지의 연혁 카드 전체 section 컴포넌트화

+ HistorySwiper.jsx file 을 생성하여, History 페이지 내에서 개발되었던 Section 전체를 컴포넌트화 하였습니다.

Related to: #6

ryudg commented on src/components/History/HistorySwiper.jsx:
```diff
@@ -0,0 +1,101 @@
+  // 연혁 내림차순으로 정렬
+  const historyDataSorted = Object.entries(historyData).toSorted((a, b) => b[0] - a[0]);
+  return (
+    <section className={styles.section}>
```
**P1**: `section` 태그를 사용한다면 `h1`~`h6` 요소를 사용해 제목이 있었으면 좋겠습니다.

참고: [MDN - section](https://developer.mozilla.org/ko/docs/Web/HTML/Element/section#%EC%82%AC%EC%9A%A9_%EC%9D%BC%EB%9E%8C)

dani-day replied:
```html
    <section className={styles.section}>
      <h1 className={styles.hidden}>오픈에스지 연혁</h1>
``` 

`section` 태그 하위에 `h1` 태그 추가하였습니다. 

```css
.hidden {
  @apply sr-only;
}
```

해당 태그는 디자인 상 보여지지 않아야하므로, screen reader 사용자만 읽을 수 있도록 tailwindcss 의 sr-only 를 추가해두었습니다.

ryudg replied:
`src/layout/Header.jsx` Header 컴포넌트에서 `<h1>` 태그를 사용해 log를 link로 활용하며,
`src/layout/RootLayout.jsx` RootLayout 컴포넌트에서 `<Header />` 컴포넌트를 불러와 사용하고 있기 때문에 모든 페이지에서 Header 컴포넌트가 공통으로 사용됩니다.

[MDN - h1-h6](https://developer.mozilla.org/ko/docs/Web/HTML/Element/Heading_Elements#%EC%82%AC%EC%9A%A9_%EC%9D%BC%EB%9E%8C) 해당 문서를 살펴보면 **"페이지 당 하나의 `<h1>`만 사용하세요. 여러 개를 써도 오류는 나지 않겠지만, 단일 `<h1>`이 모범 사례로 꼽힙니다."** 를 반영한다면 `h1` 태그보다 하위 heading 태그 `h2`~`h6`를 사용하는게 좋아보입니다.

ryudg reviewed (changes requested):
`src/components/History/HistorySwiper.jsx` 파일의 comment 내용을 수정해 주세요

dani-day committed:
🐛 [Fix] : Section 태그 내 h1 태그 추가 및 sr-only 설정
+ Section 태그 내 제목 태그 (h1) 를 추가하였습니다.
+ 해당 태그는 sr-only 설정을 하여, 스크린 리더기에서만 제목을 읽을 수 있도록 하였습니다.

Related to: #6

dani-day committed:
🌈 [Design] : YearSlide 컴포넌트의 inActive 시 폰트 크기 변경

+ YearSlide 컴포넌트의 inActive 시 폰트 크기가 작아지도록 변경하였습니다.

Related to: #6

dani-day committed:
🐛 [Fix] : History 페이지 내 중복 사용된 h1 태그를 h2 태그로 변경

+ History 페이지 내 Header 컴포넌트에서 사용한 h1 태그가 중복되어 h2 태그로 변경하였습니다.

Related to: #6

ryudg reviewed (approved):
수정 사항 확인 했습니다.

kelvin-chihyun commented on src/components/History/History.module.css:
P2: 당장 고쳐야할 사항은 아니지만, 리팩토링 시에,
css 가독성을 위해 분류하여 저장하면 좋을 것 같습니다.
관련 내용은 컨벤션에 추가해놓겠습니다.

ex)
```css
.title {
   @apply mx-5 my-3 bg-gray-200 flex; // 레이아웃 관련
   @apply text-lg font-semibold text-red-300; // 텍스트 관련
   @apply hover:text-purple-400 md:text-sm; // 이벤트, 가상요소, 반응형 등
}
```

dani-day replied:
확인했습니다. 피드백 감사합니다 :)

kelvin-chihyun reviewed (approved):
코드 모든 부분 확인했고, 이상 없습니다.
별도로, 리팩토링 시에 고치면 좋을 의견하나 추가했습니다.
